packages/graphql-codegen-supermassive-typed-document-node-plugin/src/visitor.ts

```diff
@@ -80,104 +53,47 @@ export class TypeScriptDocumentNodesVisitor extends ClientSideBaseVisitor<
-    }
-    return this._render(
-      node,
-      this.config.supermassiveDocumentNodeOutputType !== "NONE",
-    );
+    const tagImport = this._generateImport(
+      { moduleName: "@msteams/frameworks-apollo", propName: "runtimeParse" },
+      "runtimeParse",
+      false,
+    ) as string;
+    this._imports.add(tagImport);
   }
 
-  protected _render(
+  protected _gql(
     node: FragmentDefinitionNode | OperationDefinitionNode,
-    annotate = false,
   ): string {
-    const supermassiveNode = this.addTypesToRequestDocument(this._schema, {
-      kind: Kind.DOCUMENT,
-      definitions: [node],
-    }).definitions[0] as FragmentDefinitionNode | OperationDefinitionNode;
-
-    const fragments = this._transformFragments(supermassiveNode);
-
+    const fragments = this._transformFragments(node);
     const doc = this._prepareDocument(`
     ${
-      print(supermassiveNode)
+      print(node)
         .split("\\")
         .join("\\\\") /* Re-escape escaped values in GraphQL syntax */
     }
-    ${this._includeFragments(fragments)}`);
-
-    if (this.config.documentMode === DocumentMode.documentNode) {
-      let gqlObj = gqlTag([doc]);
-
-      if (this.config.optimizeDocumentNode) {
-        gqlObj = optimizeDocumentNode(gqlObj);
-      }
-      if (annotate) {
-        gqlObj = this._transformDocumentNodeToSupermassive(gqlObj);
-      }
-
-      return JSON.stringify(gqlObj);
-    } else if (
-      this.config.documentMode === DocumentMode.documentNodeImportFragments
-    ) {
-      let gqlObj = gqlTag([doc]);
-
-      if (this.config.optimizeDocumentNode) {
-        gqlObj = optimizeDocumentNode(gqlObj);
-      }
-
-      if (fragments.length > 0) {
-        const definitions = [
-          ...gqlObj.definitions.map((t) =>
-            JSON.stringify(
-              annotate
-                ? this.addTypesToRequestDocument(this._schema, {
-                    kind: Kind.DOCUMENT,
-                    definitions: [t],
-                  }).definitions[0]
-                : t,
-            ),
-          ),
-          ...fragments.map((name) => `...${name}.definitions`),
-        ].join();
-
-        return `{"kind":"${Kind.DOCUMENT}","definitions":[${definitions}]}`;
-      }
-      if (annotate) {
-        gqlObj = this._transformDocumentNodeToSupermassive(gqlObj);
-      }
-
-      return JSON.stringify(gqlObj);
-    } else if (this.config.documentMode === DocumentMode.string) {
-      return "`" + doc + "`";
-    }
-
-    const gqlImport = this._parseImport(this.config.gqlImport || "graphql-tag");
-
-    return (gqlImport.propName || "gql") + "`" + doc + "`";
+    ${this._includeFragments(fragments)}
+    `);
+
+    const gqlObj = parse(doc, { noLocation: true });
+    return `
+{
+  kind: "Document",
+  get definitions() {
+    const document = runtimeParse('${print(gqlObj).replace(/[\n\r ]+/g, " ")}');
+    delete this.definitions;
```

Comment:
This `delete` call was a hack, maybe you will be able to find a better way to save parsed state to avoid re-parsing